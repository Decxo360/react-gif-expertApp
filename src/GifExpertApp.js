import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch','Hunter x Hunter'])

    const handleAdd = (newCategory) => {
        if(categories.includes(newCategory)) return;
        setCategories([newCategory,...categories,]);
    }

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory handleAdd={event=>handleAdd(event)}></AddCategory>
            <hr></hr>


            <ol>
                {
                    categories.map((category,index) =>
                        <GifGrid
                            key={category}
                            category={category}
                        />)
                }
            </ol>
        </div>

    )
}

export default GifExpertApp