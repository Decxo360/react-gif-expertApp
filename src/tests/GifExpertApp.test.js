import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Trabajando en GifExpertApp', () => { 

    test('Debe de mostrar la snapshot', () => {

        const wrapper = shallow(<GifExpertApp></GifExpertApp>);
        expect(wrapper).toMatchSnapshot();
    });
 })