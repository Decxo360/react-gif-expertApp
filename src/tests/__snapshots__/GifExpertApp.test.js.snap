// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trabajando en GifExpertApp Debe de mostrar la snapshot 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</div>
`;
